cmd/monaco/supportarchive/supportarchive.go

```diff
@@ -29,11 +30,39 @@ import (
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/zip"
 )
 
-type SupportArchive struct {
-	Value bool
+type ctxKey struct {
+	value bool
 }
 
-func Archive(fs afero.Fs) error {
+// NewContextWithSupportArchive creates a new child context that has a signal that the support archive is required.
+// If the command has already been called, it returns the given context.
+func NewContextWithSupportArchive(ctx context.Context) context.Context {
+	if ctx.Value(ctxKey{}) != nil {
+		return ctx
+	}
+
+	key := &ctxKey{}
+	return context.WithValue(ctx, ctxKey{}, key)
```

Comment:
Please have a look at the articles I sent again.
This solution is better, but still not where it should be. 

Why even use pass struct?

Reply:
Now two different struct are used, one as key, second as value holder

Reply:
This is outdated